### DISCUSSION TITLE
Recommendation for XML converter on Android

### DISCUSSION THREAD

**Discussion by p-fischer** [square/retrofit#3732] [Q&A] [answered]
I'm just starting a green-field project. However it needs to consume an API spitting out XML.

Looking into the options which XML converter can be hooked into retrofit this is my current list of candidates:

- [Simple XML](https://github.com/square/retrofit/tree/master/retrofit-converters/simplexml) - currently 1st party converter but deprecated
- [TikXML](https://github.com/Tickaroo/tikxml) - recommended here in issues around 2017, but no commits since 2020
- [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization) - definitely future proof, no 1st party support for XML but allows serializers to be plugged in

Pluggable XML serializers for kotlinx.serialization:
- [XMLUtil](https://github.com/pdvrieze/xmlutil) - under active development, don't know why version wasn't bumped to 1.x
- [Kotlin Serialization XML](https://github.com/rharter/kotlinx-serialization-xml) - looks promising, however in such an early stage only snapshot releases are available, which means unreliable builds

(Or... sorry I realize this is a truly stupid question to ask here: Use Ktor Client, which seems to have 1st party support for [XML](https://ktor.io/docs/serialization-client.html#register_xml))

I haven't look deep into these options yet.

My wishes would be
- Refletionless serialization would be greatly preferred (tired of runtime errors in release builds due to missing proguard rules).
- The API I'm dealing with doesn't seem to use XML attributes at all but uses child tags. It would be nice if properties could default to child tags instead of XML attributes.
- Default/concise annotations so there is little effort in case the converter will be switched (e.g. to start with XMLUtil and switch to Kotlin Serialization XML later)

I'm happy to hear any opinion on this.

**Comment by JakeWharton** [comment, marked as answer]
I've been using kotlinx.serialization with XMLUtil for a year or so and it works great. You could also use Jackson which I've done in the past.

**Reply by p-fischer** [reply]
Thanks! I'm going to try that out.
